soroban-test-wasms/wasm-workspace/errors/src/lib.rs

```diff
@@ -0,0 +1,50 @@
+#![no_std]
+use soroban_sdk::{contract, contractimpl, contracterror, Val, Error};
+
+#[contract]
+pub struct Contract;
+
+#[contracterror]
+#[derive(Copy,Clone)]
+pub enum Eek
+{
+    BADNESS = 12345
```

Comment:
Somewhat unrelated, but will this work with u32::Max?

Also a bit weird that there is no `repr(u32)` here, I thought it's required...

Reply:
`u32::MAX` as an expression doesn't work -- it wants a literal. If I put `0xffff_ffff` in, it doesn't like it as the default repr choice is `isize` which is `i32` on `wasm32-unknown-unknown` and this value doesn't quite fit. If I put in `repr(u32)` it works.

In general `repr(u32)` isn't required for `#[contracterror]`, it just generates conversions to and from `Error` and `Val` that switch on the enumerated values -- so long as the enumerated values fit in the repr in question, `#[contracterror]` doesn't care (and if you provide enumerated values that _don't_ fit in the repr chosen it simply won't compile). So like .. if you _want_ to push right up against against the maximum range by using `0xffff_ffff` then yeah you'd need `repr(u32)`. But only if you want to do that.